### ISSUE TITLE
Normalizing flows training with negative loss

### ISSUE THREAD

**Issue by RayUCF** [RayUCF/NF#1]
@yarinbar. Hi, I saw you published a question '[Negative -log likelihood loss when training Normalizing Flow](https://stackoverflow.com/questions/76935527/negative-log-likelkhood-loss-when-training-normalizing-flow)'. Can you please let me know if this problem is solved?  I have the same problem. With the same code applied to different datasets, one is positive, but another is negative in terms of loss function.

**Comment by yarinbar** [comment]
In my understanding, since the log likelihood is of a *distribution* function, the value of `p(x)` may be higher than 1 - thus making the log in the negatives.

For example, a uniform distribution in [0, 0.1] will yield `p(x)=10`.

**Comment by RayUCF** [comment]
Hello Yarin,

Thank you so much for the reply.

From my understanding, the loss function presents the KL divergence. With a
negative natural logarithm, it should be positive, right?

If the loss function toward to negative infinity, it is hard to determine
when to stop training process.

Best,
Ray
